packages/stripe-graphql/README.md

```diff
@@ -0,0 +1,141 @@
+<h1 align="center">@nhost/stripe-graphql</h1>
+<h2 align="center">Stripe GraphQL</h2>
+
+<p align="center">
+  <img alt="npm" src="https://img.shields.io/npm/v/@nhost/stripe-graphql">
+  <img alt="npm" src="https://img.shields.io/npm/dm/@nhost/stripe-graphql">
+  <a href="LICENSE">
+    <img src="https://img.shields.io/badge/license-MIT-yellow.svg" alt="license: MIT" />
+  </a>
+</p>
+
+Stripe GraphQL API for Hasura Remote schemas.
+
+Connect data in your database with data from Stripe, via GraphQL.
+
+```graphql
+query {
+  users {
+    # User in your database
+    id
+    displayName
+    userData {
+      stripeCustomerId # Customer's Stripe Customer Id
+      stripePaymentMethods {
+        # Data directly from Stripe
+        id
+        card {
+          brand
+          last4
+        }
+      }
+    }
+  }
+}
+```
+
+## Install
+
+```bash
+npm install @nhost/stripe-graphql
+```
+
+## Quick Start
+
+### Serverless Function Setup
+
+Inside a Nhost project, create a file in `functions/graphql/stripe.ts`:
+
+```js
+import Stripe from 'stripe'
+
+import { createStripeGraphQLServer } from '@nhost/stipe-graphql'
+
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY!, {
+  apiVersion: '2022-08-01'
+})
+
+const server = createStripeGraphQLServer({
+  context: () => {
+    return { stripe }
+  }
+})
+
+export default server;
+```
+
+### Test
+
+Test the Stripe GraphQL API:
+
+[http://localhost:1337/v1/functions/graphql/stripe](http://localhost:1337/v1/functions/graphql/stripe)
+
+### Remote Schema
+
+Add the Stripe GraphQL API as a Remote Schema in Hasura.
+
+URL: `{{NHOST_BACKEND_URL}}/v1/functions/graphql/stripe`
+
+## Context
+
+You need to provide a `context` function to the Stripe GraphQL Server.
+
+You must return a `stripe` object and optional (recommended) `allowedStripeCustomerIds`.
+
+Minimal example:
+
+```js
+const server = createStripeGraphQLServer({
+  context: () => {
+    return { stripe }
+  }
+})
+```
+
+Realistic example:
+
+```js
+const server = createStripeGraphQLServer({
+  context: ({ request }) => {
+
+  const authorizationHeader = request.headers.get('Authorization');
+  const accessToken = authorizationHeader?.split(' ')[1];
+
+  const userFromAccessToken = accessToken
+    ? getUserFromAccessToken(accessToken)
+    : undefined;
+
+  if (!userFromAccessToken) {
+    return { stripe };
+  }
+
+  // get user's stripe customer ids
+  const { user } = await gqlSDK.getUser({
+    id: userFromAccessToken.id,
+  });
+
+  if (!user) {
+    return { stripe };
+  }
+
+  // get allowed stripe customer ids for this customer
+  const allowedStripeCustomerIds = user?.workspaceMembers
+    .filter((wm) => {
+      return typeof wm.workspace.stripeCustomerId === 'string';
+    })
+    .map((wm) => {
+      return wm.workspace.stripeCustomerId as string;
+    });
+
+    return { stripe, allowedStripeCustomerIds }
```

Comment:
- `stripe` did not/should not change in this method. And even though it would have changed, its reference wouldn't have. Hence the idea to set `stripe` as a parameter on server creation
- From what I understand, the purpose of this method is to answer to the question: is the current requested authorised to get this information? I would simplify its signature to `boolean`.

Reply:
- Stripe is no longer required here.
- Not really, `allowedStripeCustomerIds` is an array of allowed stripe customer ids with which users should be allowed to interact.